posts/a-primer-on-snowflake-stored-procedures.md

```diff
@@ -0,0 +1,205 @@
+> ```plsql
+> SET SESSION_VAR1 = 'some value';
+> CALL MyStoredProcedure(); -- Executes READ_SESSION_VAR1 and uses the value of SESSION_VAR1
+>                           -- Executes SET_SESSION_VAR2, which is still set after the procedure all is completed.
+> SELECT * FROM table WHERE column1 = $SESSION_VAR2;
+> ```
+> This is equivalent to executing the following statements:
+> ``` plsql
+> SET SESSION_VAR1 = 'some value';
+> READ_SESSION_VAR1;
+> SET_SESSION_VAR2;
+> SELECT * FROM table WHERE column1 = $SESSION_VAR2;
+> ```
+>
+> For more examples, refer to [Session Variables with Caller's Rights and Owner's Rights](https://docs.snowflake.net/manuals/sql-reference/stored-procedures-usage.html#label-sp-session-variables-example).
+
+In many cases, this is the desired behavior since we may want to inherit contextual information. In cases where the stored procedure should be more isolated, Snowflake provides a few pointers:
+* Avoid using session-level variables directly. Pass them as explicit parameters, as it forces the caller to think about exactly what values the stored procedure will use.
+* Clean up any session-level variables that you set inside the stored procedure and use names that are not likely to be used anywhere else.
+Above all, remember that unlike programming in languages such as C or Java, variables set inside a stored procedure will not be garbage collected once the function is finished. Isolating your stored procedure from its environment requires more effort in SQL than in C or Java.
+
+### Owner's Rights Stored Procedures
+Owner's Rights Stored Procedures adhere to the following rules in a session:
+* Run with the privileges of the owner.
+* Inherit the current warehouse of the caller.
+* Use the database and schema that the stored procedure is created in.
+* Cannot access most caller-specific information ^[This information includes viewing, setting and unsetting session variables, calling functions such as CURRENT_USER() and querying INFORMATION_SCHEMA table functions that return results based on the current user.]
+* Can only read supported session parameters, and cannot set or unset any of the caller’s session parameters.
+* Non-owners are not able to view information about the procedure from the PROCEDURES view.
+
+### Choosing between Owner's Rights and Caller's Rights
+Create a stored procedure as an owner's rights stored procedure if **all** of the following is true:
+* A task needs to be delegated to another user who will run with owner's privileges, not their own.
+  * Ex: If you need a user without DELETE privilege on a table to be able to call a procedure to delete old data, without manipulating current data.
+* The restrictions on Owner's Rights Stored Procedures (discussed in the section above) will not prevent a procedure from working as intended.
```

Comment:
Is there another reason that can be mentioned here? Or an example given? Seems understood already otherwise.